components/ModEditSubmissionForm.vue

```diff
@@ -604,86 +609,216 @@ function autofillEditSubmissionForm(submissionData: Submission | undefined) {
                 case 'googleMapsUrl':
                     googlemapsURL.value = submissionData['facility']?.contact?.googleMapsUrl || submissionData['googleMapsUrl']
                     break
+                case 'healthcareProfessionals':
+                    healthCareProfessionalNameArray.value = submissionData?.healthcareProfessionals?.[0]?.names
+                    ?? [{
+                        firstName: submissionData?.healthcareProfessionalName?.split(' ')[0] || '',
+                        lastName: submissionData?.healthcareProfessionalName?.split(' ')[1] || '',
```

Comment:
Right now both the first and last name are necessary for the submission. This is just to autofill from a submission the doctor. There is no middle name field on the submission form from a user either. It defaults to an empty string if undefined. I would think that would be ok for now. Also this is just for a user submitted doctor so no matter what, the moderator needs to do the work of making sure this name is correct or even exists. That is how I understood the intent? If we do have ones with middle names or last names only. This could be updated, but it still falls on the moderator to confirm.

Reply:
it's technically fine as is right now and we'd probably be fine when the data changes since our team is small and we understand our code. The risk increases when you can't track what needs to change when your data/API does (example: middle name would silently break this name split logic since it won't throw an error, but it will start showing in last name field). To prevent this, **the recommended practice is to always assume your data coming from the API could be changed/missing/different format**. 
If/when that happens, you want the UI to at least not crash completely. 
Usually there's a UX decision on showing partial successful rendering or friendly crashing the page and redirecting the user. 

An alternative approach is you do all of the proper data integrity checks when you receive data from the API instead of when it's used. 

I leave it up to you if you want to change it now, in a future ticket, or just not worry about it until we hit that circumstance 😸

Reply:
I just want some way to display the name that was entered by the submission. But I really do not understand what you mean. If it shows the middle name instead of the last name. It is still showing something and no matter what the moderator has to check that the submission is a real one. I can check and validate and check the length and render based on that. Does that make sense @ermish.

Reply:
I did it!